terraform/basic_components/main.tf

```
@@ -20,6 +20,16 @@ module "common" {
   security_group_name = var.aoc_vpc_security_group
 }
 
+locals {
+  otconfig_path = fileexists("${var.testcase}/otconfig.tpl") ? "${var.testcase}/otconfig.tpl" : module.common.default_otconfig_path
+
+  subnet_ids_list = tolist(data.aws_subnet_ids.aoc_public_subnet_ids.ids)
+
+  subnet_ids_random_index = random_id.subnetSelector.dec % length(data.aws_subnet_ids.aoc_public_subnet_ids.ids)
```

```suggestion
  subnet_ids_random_index = random_id.subnetSelector.dec % length(subnet_ids_list)
```

Might as well use the list you just made.